### PR TITLE
Upgrade GitLab to 10.4.0

### PR DESCRIPTION
**Pull request by solidnerd**

### CODE REVIEW

**Comment by pieterlange** [comment]
https://about.gitlab.com/2018/01/22/gitlab-10-4-released/

**Comment by innuendonh** [comment]
Does this update include fixes introduced in https://about.gitlab.com/2018/01/23/gitlab-10-dot-3-dot-6-released/?
In would be very nice to have 10.3.6 merged here before 10.4.....

**Comment by mathewpeterson** [comment]
@innuendonh It would not, each Gitlab release has its own PR here, in order to keep a clean git history. Somebody will need to create a PR for 10.3.6.

**Comment by solidnerd** [comment]
@innuendonh  You are lucky. I think it's a good idea to release the latest version 10.3.6 until we gonna go for 10.4. In 10.4 I plan now to use ubuntu xenial over trusty for long support. I will update all branches today / tomorrow. I think we could plan with 10.4 tomorrow.

**Comment by cwildfoerster** [comment]
@solidnerd Nice, glad to hear you're upgrading to xenial. Thank you very much for your work!

**Comment by sameersbn** [comment]
@solidnerd i've released sameersbn/ubuntu:16.04.20180124. Maybe we can have it in this release?

**Comment by solidnerd** [comment]
@sameersbn It's already in there https://github.com/sameersbn/docker-gitlab/pull/1479/commits/33427dac44e586bd9a2b74fd92d25c0511968e4d

**Comment by sameersbn** [comment]
> @sameersbn It's already in there 33427da

oh sweet!

**Comment by Ilya-Kuchaev** [comment]
Any chance to see it merged shortly?

**Comment by LennardWesterveld** [comment]
Please also update golang to Go 1.9.2 see > https://about.gitlab.com/2018/01/22/gitlab-10-4-released/#omnibus-improvements

**Comment by solidnerd** [comment]
@LennardWesterveld I added Golang 1.9.3  in e702071 . 

@Ilya-Kuchaev I think I could resolve all problems now. I'm gonna now to test it if it's working correctly.

File: Changelog.md [hunk 1]
```diff
@@ -3,6 +3,9 @@
 This file only reflects the changes that are made in this image. Please refer to the upstream GitLab [CHANGELOG](
 https://gitlab.com/gitlab-org/gitlab-ce/blob/master/CHANGELOG.md) for the list of changes in GitLab.
 
+**10.4.0**
+- gitlab: upgrade CE to v10.4.0
```
Comment (drubin) [inline Changelog.md]: @solidnerd Please also include that you replaced base Docker image with `sameersbn/ubuntu:16.04.20180124` which upgrades to ubuntu 16.04 from 14.04

File: assets/runtime/config/gitaly/config.toml [hunk 2]
```diff
@@ -39,6 +39,16 @@ path = "{{GITLAB_REPOS_DIR}}"
 # The directory where gitaly-ruby is installed
 dir = "/home/git/gitaly/ruby"
 
+# # Gitaly-ruby resident set size (RSS) that triggers a memory restart (bytes)
```
[the remark this answers is not in the document]
Comment (solidnerd) [reply assets/runtime/config/gitaly/config.toml]: I didn't these configuration options. They will be added  it was during the update from the upstream.

**Comment by drubin** [comment]
Is there anything outstanding? 

Do you need help with testing this?

**Comment by solidnerd** [comment]
@drubin everything is fine. I'm gonna start with the release now.

**Comment by solidnerd** [comment]
LGTM !

**Pull request merged**